File: design-patterns/ts/chain-of-responsibility/interview/hr-interview.ts
// hr-interview.ts

import Interview from "./interview";

class HrInterview extends Interview {
    constructor(nextInterview: Interview | null) {
        super(nextInterview);
    }

    execute(): void {
        // Ask all questions for hr interview
        // Perform any other action required for hr interview
        console.log("Ask hr interview questions")

        // Execute the next interview set while creating new struct
        if (this.nextInterview) {
            this.nextInterview.execute();
        }
    }

}

export default HrInterview;

File: design-patterns/go/chain_of_responsibility/interview/hr_interview.go
// hr_interview.go

package main

import "fmt"

type HrInterview struct {
	nextInterview Interview
}

func NewHrInterview(nextInterview Interview) (hrInterview *HrInterview) {
	hrInterview = &HrInterview{}
	hrInterview.nextInterview = nextInterview
	return
}

func (hrInterview *HrInterview) Execute() {
	// Ask all questions for Hhr interview
	// Perform any other action required for Hhr interview
	fmt.Println("Ask HR interview questions")

	// Execute the next interview set while creating new struct
	if hrInterview.nextInterview != nil {
		hrInterview.nextInterview.Execute()
	}
}
